### PR TITLE
Fix C-API Compilation Error

### PR DESCRIPTION
**Pull request by gongweibao**
FIX https://github.com/PaddlePaddle/Paddle/issues/3329

### CODE REVIEW

File: paddle/framework/CMakeLists.txt [hunk 1]
```diff
@@ -44,3 +46,4 @@ cc_library(paddle_pybind SHARED
 	mean_op
 	cross_entropy_op
 	recurrent_op)
+endif()
```
Comment (typhoonzero) [inline paddle/framework/CMakeLists.txt]: `endif()` => `endif(WITH_PYTHON)`

File: paddle/framework/CMakeLists.txt [hunk 2]
```diff
@@ -35,6 +35,8 @@ add_dependencies(framework_py_proto framework_py_proto_init)
 
 cc_library(backward SRCS backward.cc DEPS net_op)
 cc_test(backward_test SRCS backward_test.cc DEPS backward)
+
+if (WITH_PYTHON)
```
Comment (typhoonzero) [inline paddle/framework/CMakeLists.txt]: No space after if in cmake

**Comment by wangkuiyi** [comment]
Fixcapi ==> Fix C-API Compilation Error

**Comment by wangkuiyi** [comment]
Also, we need a Github issue describing the problem and the ideal to fix it. A good example is @luotao1 's work https://github.com/PaddlePaddle/Paddle/pull/3310

File: paddle/framework/CMakeLists.txt (hunk 2 above)
Comment (gongweibao) [inline paddle/framework/CMakeLists.txt]: Done.

**Comment by gongweibao** [comment]
Thanks all!

**Pull request merged**